lib/improver/tests/utilities/test_OrographicAlphas.py

```diff
@@ -0,0 +1,188 @@
+from improver.utilities.spatial import DifferenceBetweenAdjacentGridSquares
+
+
+def set_up_cube():
+    """Set up dummy cube for tests"""
+    data = np.array([[1., 5., 10.],
+                     [3., 4., 7.],
+                     [0., 2., 1.]])
+    cube = Cube(data, "precipitation_amount", units="kg m^-2 s^-1")
+    cube.add_dim_coord(DimCoord(np.linspace(0.0, 4.0, 3), 'latitude',
+                                units='m'), 0)
+    cube.add_dim_coord(DimCoord(np.linspace(0.0, 4.0, 3), 'longitude',
+                                units='m'), 1)
+    return cube
+
+
+class Test__init__(IrisTest):
+    """Test the init method."""
+
+    def test_basic(self):
+        """Test default attribute initialisation"""
+        result = OrographicAlphas()
+        self.assertTrue(result.invert_alphas)
+        self.assertEqual(result.min_alpha, 0.)
+        self.assertEqual(result.max_alpha, 1.)
+        self.assertEqual(result.coefficient, 1.)
+        self.assertEqual(result.power, 1.)
+
+
+class Test__repr__(IrisTest):
+    """Test the repr method."""
+
+    def test_basic(self):
+        """Test that the __repr__ returns the expected string."""
+        result = str(OrographicAlphas())
+        msg = ('<OrographicAlphas: min_alpha: {}; max_alpha: {}; '
+               'coefficient: {}; power: {}; '
+               'invert_alphas: {}>'.format(0.0, 1.0, 1, 1, True))
+        self.assertEqual(result, msg)
+
+
+class Test_scale_alphas(IrisTest):
+    """Class to test the scale_alphas function"""
+
+    def setUp(self):
+        """Set up cube & plugin"""
+        self.plugin = OrographicAlphas()
+        cube = set_up_cube()
+        self.cubelist = [cube, cube]
+
+    def test_basic(self):
+        """
+        Test the basic function of scale_alphas, using the
+        standard max and min alphas.
+        """
+        result = self.plugin.scale_alphas(self.cubelist)
+        expected = np.array([[0.1, 0.5, 1.0],
+                             [0.3, 0.4, 0.7],
+                             [0.0, 0.2, 0.1]])
+        self.assertArrayAlmostEqual(result[0].data, expected)
+        self.assertArrayAlmostEqual(result[1].data, expected)
+
+    def test_maxmin(self):
+        """
+        Tests the function of scale_alphas, using a max
+        and min value for alpha.
+        """
+        result = self.plugin.scale_alphas(self.cubelist, 0.3, 0.5)
+        expected = np.array([[0.32, 0.40, 0.50],
+                             [0.36, 0.38, 0.44],
+                             [0.30, 0.34, 0.32]])
+        self.assertArrayAlmostEqual(result[0].data, expected)
+        self.assertArrayAlmostEqual(result[1].data, expected)
+
+
+class Test_unnormalised_alphas(IrisTest):
+    """Dummy class to test the basic alphas function (currently no effect)"""
+
+    def setUp(self):
+        """Set up cube & plugin"""
+        self.plugin = OrographicAlphas()
+        self.cube = set_up_cube()
+
+    def test_basic(self):
```

Comment:
This isn't really a test. This tests that numbers generated with DifferenceBetweenAdjacentGridSquares are unchanged by being copied into a new cube with no coefficient or power applied. This will also fail once the abs() discussed below is moved.

Reply:
Agree this isn't really a test at present.  I want something there as a placeholder as on principle I think that if someone changes things (eg the default coefficients) then the unit tests should break.  And because this might get more complex in the future.  However I can remove this if preferred.